va/sysdeps.h

```diff
@@ -48,7 +48,7 @@
 # include <log/log.h>
 #endif
 
-#if defined __GNUC__ && defined HAVE_GNUC_VISIBILITY_ATTRIBUTE
+#if defined __GNUC__ && defined HAVE_GNUC_VISIBILITY_ATTRIBUTE && !defined(__MINGW32__)
```

Comment:
@eli-schwartz: can you, please, comment on this? are you ok with this change considering wrong detection of visibility attribute for mingw?

Reply:
I object to the use of HAVE_GNUC_VISIBILITY_ATTRIBUTE.

I'm -0 with checking for `__MINGW32__`, it seems overly specific to the mingw case when really this is about Windows, but it probably doesn't matter if you don't support building with MSVC...

Reply:
> but it probably doesn't matter if you don't support building with MSVC

We do support building with MSVC. I afraid I don't understand what's your proposal to handle this case. We do have `cc.has_argument` already and it does not seem to work correctly:
```
# Symbol visibility
if cc.has_argument('-fvisibility=hidden')
  va_c_args += ['-DHAVE_GNUC_VISIBILITY_ATTRIBUTE']
endif
```
See: https://github.com/intel/libva/blob/da4a18fe814c9695f420d466ad3065cebdbe288a/meson.build#L124

Are you saying we should not pass `HAVE_GNUC_VISIBILITY_ATTRIBUTE` at all? But how to handle that then. Please, be specific.

Reply:
> Instead of checking for i`f __GNUC__ >= 4`, we use `HAVE_GNUC_VISIBILITY_ATTRIBUTE` to check dynamically for compiler attribute support, which will also be set to false by meson when building with MSVC. 

This right here is the thing I'm suggesting to not do.

Just check for `__GNUC__` as the GCC wiki suggests. Compilers which are compatible with the GNUC visibility attribute announce that support by defining their `__GNUC__` compatibility level. You don't need to fork and exec the compiler during setup in order to check that -- the preprocessor can do it much more efficiently while the source code is being parsed.

This makes your source code more reliable (less context depends on the build system, less things you need to implement multiple times -- once in meson.build and once in configure.ac) and speeds up builds. It's a win-win.

Why do you need to run configure-time checks? What problem are you trying to solve?

The problem in question isn't Windows support, because not only does MSVC not define `__GNUC__`, but you can just check `_WIN32` anyway, which works on mingw too.

Reply:
Thank you for clarification. This makes sense to me.

Reply:
If you implement my suggestion, you can drop 22 lines from configure.ac right here:

https://github.com/intel/libva/blob/7afc9aea5b350a0ed114969c98f6449dce49ded5/configure.ac#L190-L211

Reply:
@eli-schwartz thanks for the clarification, I agree it's a simpler and better way. I've just pushed changes with it, please let me know what you think.

Reply:
It.looka very nice now. :D